HW3/P4/median_filter.cl

```diff
@@ -21,14 +113,38 @@ median_3x3(__global __read_only float *in_values,
     //
     // Note that globally out-of-bounds pixels should be replaced
     // with the nearest valid pixel's value.
+        if (idx_1D < buf_w) {
+            for (row = 0; row < buf_h; row++) {
+                buffer[row * buf_w + idx_1D] = \
+                    fetch_point(in_values, w, h,
+                          buf_corner_x + idx_1D,
+                          buf_corner_y + row);
+          }
+        }
 
+        barrier(CLK_LOCAL_MEM_FENCE);
```

Comment:
A barrier should be performed by all threads (not only the ones that enter the if statement).